File: 백준/Bronze/33897. 헤키레키잇센/헤키레키잇센.py
import sys

n = sys.stdin.readline()
arr = list(map(int, input().split()))

maxContinue = 0
tempContinue = 0
useSkillCnt = 1
preNum = arr[0]

for i in arr:
    if i < preNum:
        if tempContinue > maxContinue: maxContinue = tempContinue
        useSkillCnt += 1
        tempContinue = 1
    else:
        tempContinue += 1
    preNum = i

if tempContinue > maxContinue: maxContinue = tempContinue
print(str(useSkillCnt) + ' ' + str(maxContinue))

File: 백준/Bronze/33897. 헤키레키잇센/헤키레키잇센.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int[] arr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();

        int maxContinue = 0;
        int tempContinue = 0;
        int skillCnt = 1;
        int preNum = arr[0];

        for (int i : arr) {
            if(preNum > i) {
                if(tempContinue > maxContinue) maxContinue = tempContinue;
                skillCnt++;
                tempContinue = 1;
            }
            else{
                tempContinue++;
            }
            preNum = i;
        }

        if(tempContinue > maxContinue) maxContinue = tempContinue;
        System.out.println(skillCnt + " " + maxContinue);
    }
}
